cmd/bb_remote_asset/main.go

```diff
@@ -45,9 +46,14 @@ func main() {
 	// Initialize CAS storage access
 	grpcClientFactory := bb_grpc.DefaultClientFactory
 
-	assetStore, contentAddressableStorage, err := configuration.NewAssetStoreAndCASFromConfiguration(config.AssetCache, grpcClientFactory, int(config.MaximumMessageSizeBytes))
+	contentAddressableStorageInfo, err := blobstore_configuration.NewBlobAccessFromConfiguration(config.ContentAddressableStorage, blobstore_configuration.NewCASBlobAccessCreator(grpcClientFactory, int(config.MaximumMessageSizeBytes)))
 	if err != nil {
-		log.Fatalf("Failed to create asset store and CAS: %v", err)
+		log.Fatalf("Failed to create CAS blob access: %v", err)
+	}
+
+	assetStore, err := configuration.NewAssetStoreFromConfiguration(config.AssetCache, contentAddressableStorageInfo, grpcClientFactory, int(config.MaximumMessageSizeBytes))
```

Comment:
Rather the check for `config.AssetCache != nil` should be here to avoid the need for `NewAssetStoreFromConfiguration` to return `nil, nil`